### DISCUSSION TITLE
Safe resize / and crop with aspect ratio keeping.

### DISCUSSION THREAD

**Discussion by codec-abc** [SixLabors/ImageSharp#2790] [Q&A] [answered]
I am writing a software to easily resize and crop image to a specific resolution. I managed to do the GUI in a Winform app but I struggle to do the resize and crop. What I would like is to give a rectangle where any corner can be outside the image (meaning x or y can be < 0 or x >= width or y >= height) without having an exception. It would be even better if I can easily specify the color to use to fill the empty space.  
An image being clearer, let's say I want to extract the area defined by the red rectangle and resize it to a specific resolution while controlling the color of the black area. Is there an easy way to do this out the box?

![image](https://github.com/user-attachments/assets/ee141448-6aaf-4aa1-b2d3-d3109e7f4adc)

For the record Crop and Resize but did not manage to do this.

**Comment by JimBobSquarePants** [comment, marked as answer]
Hi @codec-abc 

Yes you should be able to do this with the Resize Options. https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Processing.ResizeOptions.html?q=ResizeOptions

I would first crop using an intersection of the image bounds and the crop rectangle, you can then do a resize using the pad mode and set a pad color.

**Comment by codec-abc** [comment]
@JimBobSquarePants Works like a charm. You probably saved me a few hours of toying around. Thanks!

**Reply by JimBobSquarePants** [reply]
Great to hear!
